pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/ClusterTest.java

```diff
@@ -91,54 +88,66 @@ protected void startBroker() {
     startBrokers(1);
   }
 
-  protected void startBrokers(int brokerCount) {
-    try {
-      for (int i = 0; i < brokerCount; ++i) {
-        Configuration configuration = BrokerTestUtils.getDefaultBrokerConfiguration();
-        configuration.setProperty("pinot.broker.timeoutMs", 100 * 1000L);
-        configuration.setProperty("pinot.broker.client.queryPort", Integer.toString(BROKER_PORT + i));
-        configuration.setProperty("pinot.broker.routing.table.builder.class", "random");
-        configuration.setProperty("pinot.broker.delayShutdownTimeMs", 0);
-        overrideBrokerConf(configuration);
-        _brokerStarters.add(BrokerTestUtils.startBroker(_clusterName, ZkStarter.DEFAULT_ZK_STR, configuration));
-      }
-    } catch (Exception e) {
-      throw new RuntimeException(e);
-    }
+  protected void startBroker(int basePort, String zkStr) {
+    startBrokers(1, basePort, zkStr);
   }
 
-  protected void startServer() {
-    startServers(1);
+  protected void startBrokers(int numBrokers) {
+    startBrokers(numBrokers, DEFAULT_BROKER_PORT, ZkStarter.DEFAULT_ZK_STR);
   }
 
-  protected void startServer(Configuration configuration) {
-    startServers(1, configuration);
+  protected void startBrokers(int numBrokers, int basePort, String zkStr) {
+    _brokerBaseApiUrl = "http://localhost:" + basePort;
+    for (int i = 0; i < numBrokers; i++) {
+      Configuration configuration = BrokerTestUtils.getDefaultBrokerConfiguration();
+      configuration.setProperty("pinot.broker.timeoutMs", 100 * 1000L);
+      configuration.setProperty("pinot.broker.client.queryPort", Integer.toString(basePort + i));
+      configuration.setProperty("pinot.broker.routing.table.builder.class", "random");
+      configuration.setProperty("pinot.broker.delayShutdownTimeMs", 0);
+      overrideBrokerConf(configuration);
+      _brokerStarters.add(BrokerTestUtils.startBroker(_clusterName, zkStr, configuration));
+    }
   }
 
-  public Configuration getDefaultConfiguration() {
+  public static Configuration getDefaultServerConfiguration() {
     Configuration configuration = DefaultHelixStarterServerConfig.loadDefaultServerConf();
-    configuration.setProperty(Helix.KEY_OF_SERVER_NETTY_HOST, NetUtil.getHostnameOrAddress());
+    configuration.setProperty(Helix.KEY_OF_SERVER_NETTY_HOST, LOCAL_HOST);
```

Comment:
Since we are always running the test locally, we can always set the hostname to `localhost`.